kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala

```diff
@@ -80,7 +80,8 @@ class JDBCMetadataStore(conf: KyuubiConf) extends MetadataStore with Logging {
   hikariConfig.setPoolName("jdbc-metadata-store-pool")
 
   @VisibleForTesting
-  implicit private[kyuubi] val hikariDataSource = new HikariDataSource(hikariConfig)
+  implicit private[kyuubi] val hikariDataSource: HikariDataSource =
+    new HikariDataSource(hikariConfig)
```

Comment:
```suggestion
  implicit private[kyuubi] val hikariDataSource = new HikariDataSource(hikariConfig)
```
Revert unnecessary change.

Reply:
With Scala 2.13.11 or above(2.13.9, 2.13.10 not tested), if we use previous code `implicit private[kyuubi] val hikariDataSource = new HikariDataSource(hikariConfig)`,  an error will occur during the compilation phase:

```
kyuubi/kyuubi-server/src/main/scala/org/apache/kyuubi/server/metadata/jdbc/JDBCMetadataStore.scala:83: Implicit definition should have explicit type (inferred com.zaxxer.hikari.HikariDataSource) [quickfixable]
```

Reply:
OK.